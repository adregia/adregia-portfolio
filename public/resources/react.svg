<svg width="35" height="35" viewBox="0 0 35 35" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M17.4693 20.7116C19.2291 20.7116 20.6557 19.2849 20.6557 17.5251C20.6557 15.7653 19.2291 14.3386 17.4693 14.3386C15.7094 14.3386 14.2828 15.7653 14.2828 17.5251C14.2828 19.2849 15.7094 20.7116 17.4693 20.7116Z" fill="#61DAFB"/>
<path d="M17.4693 29.2431C19.9705 31.6757 22.4374 33.1148 24.4247 33.1148C25.0757 33.1148 25.6924 32.9777 26.2064 32.6694C27.988 31.6415 28.639 29.0718 28.0223 25.4056C27.9195 24.7546 27.7825 24.1036 27.6111 23.4183C28.2964 23.2128 28.9131 23.0072 29.5299 22.8016C32.9904 21.4653 34.9092 19.6151 34.9092 17.5936C34.9092 15.5378 32.9904 13.6876 29.5299 12.3856C28.9131 12.1458 28.2964 11.9402 27.6111 11.7689C27.7825 11.0836 27.9195 10.4326 28.0223 9.78163C28.6048 6.04696 27.9538 3.44296 26.1721 2.41507C25.6581 2.1067 25.0414 1.96965 24.3904 1.96965C22.4374 1.96965 19.9362 3.4087 17.435 5.84138C14.9681 3.4087 12.5011 1.96965 10.5139 1.96965C9.86286 1.96965 9.24613 2.1067 8.73218 2.41507C6.9505 3.44296 6.2995 6.01269 6.91624 9.67884C7.01903 10.3298 7.15608 10.9808 7.3274 11.6661C6.64213 11.8717 6.0254 12.0773 5.40866 12.2828C1.94809 13.6191 0.0293579 15.4693 0.0293579 17.4908C0.0293579 19.5466 1.94809 21.3968 5.40866 22.6988C6.0254 22.9387 6.64213 23.1442 7.3274 23.3156C7.15608 24.0008 7.01903 24.6518 6.91624 25.3028C6.33377 28.969 6.98476 31.573 8.73218 32.5666C9.24613 32.875 9.86286 33.012 10.5139 33.012C12.5011 33.0805 14.9681 31.6415 17.4693 29.2431ZM15.5505 24.6175C16.1673 24.6518 16.8183 24.6518 17.4693 24.6518C18.1203 24.6518 18.7713 24.6518 19.388 24.6175C18.7713 25.4399 18.1203 26.1937 17.4693 26.9132C16.8183 26.1937 16.1673 25.4399 15.5505 24.6175ZM11.3019 21.0885C11.6445 21.6709 11.9529 22.2191 12.3298 22.7673C11.2677 22.6303 10.274 22.459 9.31465 22.2534C9.62302 21.3283 9.96565 20.3689 10.3768 19.4096C10.6509 19.9578 10.9593 20.5402 11.3019 21.0885ZM9.31465 12.831C10.274 12.6255 11.2677 12.4542 12.3298 12.3171C11.9872 12.8653 11.6445 13.4135 11.3019 13.996C10.9593 14.5785 10.6509 15.1267 10.3768 15.7091C9.93139 14.7155 9.58876 13.7562 9.31465 12.831ZM11.1991 17.5251C11.6445 16.6 12.1242 15.6749 12.6724 14.7498C13.1864 13.8589 13.7689 12.9681 14.3513 12.0773C15.3792 12.0087 16.4071 11.9745 17.4693 11.9745C18.5657 11.9745 19.5936 12.0087 20.5872 12.0773C21.2039 12.9681 21.7522 13.8589 22.2661 14.7498C22.8143 15.6749 23.294 16.6 23.7394 17.5251C23.294 18.4502 22.8143 19.3753 22.2661 20.3004C21.7522 21.1912 21.1697 22.0821 20.5872 22.9729C19.5593 23.0414 18.5314 23.0757 17.4693 23.0757C16.3729 23.0757 15.345 23.0414 14.3513 22.9729C13.7346 22.0821 13.1864 21.1912 12.6724 20.3004C12.1242 19.3753 11.6445 18.4502 11.1991 17.5251ZM24.596 15.6749L23.6709 13.9617C23.3283 13.3793 23.0199 12.831 22.643 12.2828C23.7052 12.4199 24.6988 12.5912 25.6581 12.7968C25.3498 13.7562 25.0071 14.7155 24.596 15.6749ZM24.596 19.3753C25.0071 20.3347 25.3498 21.294 25.6581 22.2191C24.6988 22.4247 23.7052 22.596 22.643 22.7331C22.9856 22.1849 23.3283 21.6367 23.6709 21.0542C23.9793 20.5402 24.2876 19.9578 24.596 19.3753ZM25.384 31.2646C25.1099 31.4359 24.7673 31.5044 24.3904 31.5044C22.7115 31.5044 20.6215 30.1339 18.5657 28.0781C19.5593 27.016 20.5187 25.8168 21.4781 24.4805C23.0884 24.3434 24.6303 24.1036 26.0693 23.761C26.2406 24.3777 26.3434 24.9944 26.4462 25.6112C26.9259 28.5235 26.549 30.6136 25.384 31.2646ZM27.1657 13.2079C31.0032 14.3044 33.2988 15.9833 33.2988 17.5251C33.2988 18.8614 31.7227 20.1976 28.9474 21.2598C28.3992 21.4653 27.7825 21.6709 27.1657 21.8422C26.7203 20.4375 26.1721 18.9984 25.4868 17.5251C26.1721 16.0518 26.7546 14.6127 27.1657 13.2079ZM24.4247 3.54575C24.8016 3.54575 25.1099 3.61428 25.4183 3.78559C26.549 4.43659 26.9601 6.49238 26.4805 9.439C26.3777 10.0215 26.2406 10.6382 26.1036 11.2892C24.6645 10.9808 23.1227 10.741 21.5123 10.5697C20.5872 9.23342 19.5936 8.03422 18.6 6.97206C20.6557 4.95054 22.7458 3.54575 24.4247 3.54575ZM19.388 10.4326C18.7713 10.3984 18.1203 10.3984 17.4693 10.3984C16.8183 10.3984 16.1673 10.3984 15.5505 10.4326C16.1673 9.61032 16.8183 8.85653 17.4693 8.13701C18.1203 8.85653 18.7713 9.64458 19.388 10.4326ZM9.5545 3.78559C9.8286 3.61428 10.1712 3.54575 10.5481 3.54575C12.227 3.54575 14.3171 4.91628 16.3728 6.97206C15.3792 8.03422 14.4199 9.23342 13.4605 10.5697C11.8501 10.7067 10.3083 10.9466 8.86924 11.2892C8.69792 10.6725 8.59513 10.0557 8.49234 9.439C8.01266 6.52664 8.38955 4.47086 9.5545 3.78559ZM7.77282 21.8422C3.93535 20.7458 1.63972 19.0669 1.63972 17.5251C1.63972 16.1888 3.21582 14.8526 5.99113 13.7904C6.53934 13.5848 7.15608 13.3793 7.77282 13.2079C8.21824 14.6127 8.76644 16.0518 9.45171 17.5251C8.76644 18.9984 8.18397 20.4717 7.77282 21.8422ZM8.49234 25.6112C8.59513 25.0287 8.73218 24.412 8.86924 23.761C10.3083 24.0693 11.8501 24.3092 13.4605 24.4805C14.3856 25.8168 15.3792 27.016 16.3728 28.0781C14.3171 30.0996 12.227 31.5044 10.5481 31.5044C10.1712 31.5044 9.86286 31.4359 9.5545 31.2646C8.38955 30.6136 8.01266 28.5235 8.49234 25.6112Z" fill="#61DAFB"/>
</svg>
