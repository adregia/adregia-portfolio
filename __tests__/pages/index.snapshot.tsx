import { render } from '@testing-library/react';
import { CssVarsProvider } from '@mui/joy';

import type { ReactNode } from 'react';

import Home from '../../src/pages/index';

// `CssVarsProvider` invokes this method, so we need to mock it.
Object.defineProperty(window, 'matchMedia', {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

const wrapper = ({ children }: { children: ReactNode }) => {
  return <CssVarsProvider>{children}</CssVarsProvider>;
};

it('renders homepage unchanged', () => {
  const { container } = render(<Home />, { wrapper });
  expect(container).toMatchSnapshot();
});
