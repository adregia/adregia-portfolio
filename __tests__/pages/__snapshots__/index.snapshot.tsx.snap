// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div>
  <main
    class="main"
  >
    <div
      class="MuiCard-root MuiCard-vertical MuiCard-variantPlain MuiCard-colorNeutral MuiCard-sizeMd card css-zqmsxh-JoyCard-root"
    >
      <div
        class="sections"
        data-first-child=""
      >
        <section
          class="css-1pt1kn1"
        >
          <div
            class="css-ocxynz"
          >
            AD
          </div>
          <div
            class="css-1r6g51n"
          >
            <a
              href="https://www.linkedin.com/in/ahmed-dregia/"
              target="_blank"
            >
              <button
                class="MuiIconButton-root MuiIconButton-variantPlain MuiIconButton-colorNeutral MuiIconButton-sizeSm css-1j0ix75-JoyIconButton-root"
                type="button"
              >
                <img
                  alt="LinkedIn"
                  data-nimg="1"
                  decoding="async"
                  height="20"
                  src="/resources/icons/linkedin.svg"
                  style="color: transparent;"
                  width="20"
                />
              </button>
            </a>
            <button
              class="MuiIconButton-root MuiIconButton-variantPlain MuiIconButton-colorNeutral MuiIconButton-sizeSm css-1j0ix75-JoyIconButton-root"
              type="button"
            >
              <img
                alt="Email"
                data-nimg="1"
                decoding="async"
                height="20"
                src="/resources/icons/email.svg"
                style="color: transparent;"
                width="20"
              />
            </button>
            <button
              class="MuiIconButton-root MuiIconButton-variantPlain MuiIconButton-colorNeutral MuiIconButton-sizeSm css-1j0ix75-JoyIconButton-root"
              type="button"
            >
              <img
                alt="ResumÃ©"
                data-nimg="1"
                decoding="async"
                height="20"
                src="/resources/icons/document.svg"
                style="color: transparent;"
                width="20"
              />
            </button>
            <button
              class="MuiIconButton-root MuiIconButton-variantPlain MuiIconButton-colorNeutral MuiIconButton-sizeSm css-1j0ix75-JoyIconButton-root"
              type="button"
            >
              <img
                alt="Phone"
                data-nimg="1"
                decoding="async"
                height="20"
                src="/resources/icons/phone.svg"
                style="color: transparent;"
                width="20"
              />
            </button>
          </div>
        </section>
        <section
          class="css-31iesr"
        >
          <div
            class="css-1vki4m"
          >
            <div
              class="css-1bie8f"
            >
              <div
                class="css-1iyoj2o"
              >
                <span
                  class="css-1nsfrpq"
                >
                  Rivian
                </span>
                <span
                  class="css-qm3h0l"
                >
                  Supply Chain Portal (Web, AWS)
                </span>
              </div>
            </div>
          </div>
          <div
            class="css-1vki4m"
          >
            <div
              class="css-1bie8f"
            >
              <div
                class="css-1iyoj2o"
              >
                <span
                  class="css-1nsfrpq"
                >
                  Toyota Racing Development
                </span>
                <span
                  class="css-qm3h0l"
                >
                  Data Aggregation/Race Prediction (Web, AWS)
                </span>
              </div>
              <a
                href="https://aws.amazon.com/solutions/case-studies/toyota-racing-development/"
                target="_blank"
              >
                <button
                  class="MuiIconButton-root MuiIconButton-variantPlain MuiIconButton-colorNeutral MuiIconButton-sizeSm css-1j0ix75-JoyIconButton-root"
                  type="button"
                >
                  <img
                    alt="Link"
                    data-nimg="1"
                    decoding="async"
                    height="16"
                    src="/resources/icons/link.svg"
                    style="color: transparent;"
                    width="16"
                  />
                </button>
              </a>
            </div>
          </div>
          <div
            class="css-1vki4m"
          >
            <div
              class="css-1bie8f"
            >
              <div
                class="css-1iyoj2o"
              >
                <span
                  class="css-1nsfrpq"
                >
                  Yapi
                </span>
                <span
                  class="css-qm3h0l"
                >
                  Online Practice Portal, Internal Tools (Web)
                </span>
              </div>
              <a
                href="https://yapiapp.com/"
                target="_blank"
              >
                <button
                  class="MuiIconButton-root MuiIconButton-variantPlain MuiIconButton-colorNeutral MuiIconButton-sizeSm css-1j0ix75-JoyIconButton-root"
                  type="button"
                >
                  <img
                    alt="Link"
                    data-nimg="1"
                    decoding="async"
                    height="16"
                    src="/resources/icons/link.svg"
                    style="color: transparent;"
                    width="16"
                  />
                </button>
              </a>
            </div>
          </div>
          <div
            class="css-1vki4m"
          >
            <div
              class="css-1bie8f"
            >
              <div
                class="css-1iyoj2o"
              >
                <span
                  class="css-1nsfrpq"
                >
                  FitRecruiters (Personal Project)
                </span>
                <span
                  class="css-qm3h0l"
                >
                  Jobs Website for Fitness Professionals (Web, Firebase)
                </span>
              </div>
              <a
                href="/fit-recruiters"
                target="_blank"
              >
                <button
                  class="MuiIconButton-root MuiIconButton-variantPlain MuiIconButton-colorNeutral MuiIconButton-sizeSm css-1j0ix75-JoyIconButton-root"
                  type="button"
                >
                  <img
                    alt="Info"
                    data-nimg="1"
                    decoding="async"
                    height="16"
                    src="/resources/icons/info.svg"
                    style="color: transparent;"
                    width="16"
                  />
                </button>
              </a>
            </div>
          </div>
          <div
            class="css-1vki4m"
          >
            <div
              class="css-1bie8f"
            >
              <div
                class="css-1iyoj2o"
              >
                <span
                  class="css-1nsfrpq"
                >
                  Hoist Powertools (Personal Project)
                </span>
                <span
                  class="css-qm3h0l"
                >
                  Utilities and Resources for Powerlifters (iOS, Firebase)
                </span>
              </div>
              <a
                href="/hoist-powertools"
                target="_blank"
              >
                <button
                  class="MuiIconButton-root MuiIconButton-variantPlain MuiIconButton-colorNeutral MuiIconButton-sizeSm css-1j0ix75-JoyIconButton-root"
                  type="button"
                >
                  <img
                    alt="Info"
                    data-nimg="1"
                    decoding="async"
                    height="16"
                    src="/resources/icons/info.svg"
                    style="color: transparent;"
                    width="16"
                  />
                </button>
              </a>
            </div>
          </div>
        </section>
      </div>
      <footer
        class="css-d0ckhb"
      >
        <img
          alt="React"
          data-nimg="1"
          decoding="async"
          height="32"
          src="/resources/react.svg"
          style="color: transparent;"
          width="32"
        />
        <img
          alt="Python"
          data-nimg="1"
          decoding="async"
          height="32"
          src="/resources/python.svg"
          style="color: transparent;"
          width="32"
        />
        <img
          alt="JavaScript"
          data-nimg="1"
          decoding="async"
          height="32"
          src="/resources/js.svg"
          style="color: transparent;"
          width="32"
        />
        <img
          alt="AWS"
          data-nimg="1"
          decoding="async"
          height="32"
          src="/resources/aws.svg"
          style="color: transparent;"
          width="32"
        />
      </footer>
    </div>
  </main>
</div>
`;
